src/profile/components/multiItemAddressRow/MultiItemAddressRow.tsx

```diff
@@ -0,0 +1,189 @@
+  const countryList = countries.getNames(applicationLanguage);
+  const countryOptions = Object.keys(countryList).map(key => ({
+    value: key,
+    label: countryList[key],
+  }));
+  const formFields = getFormFields(dataType);
+  const {
+    isNew,
+    isEditing,
+    actionHandler,
+    currentAction,
+  } = useCommonEditHandling(props);
+  const { hasFieldError, getFieldErrorMessage } = createFormFieldHelpers<
+    FormikValues
+  >(t, isNew);
+
+  const { primary } = data;
+
+  if (isEditing) {
+    return (
+      <Formik
+        initialValues={{
+          address,
+          city,
+          postalCode,
+          countryCode,
+        }}
+        onSubmit={async values => {
+          await actionHandler('save', values);
+        }}
+        validationSchema={addressSchema}
+      >
+        {(formikProps: FormikProps<FormikValues>) => (
+          <Form className={commonFormStyles.multiItemForm}>
+            <h4 className={commonFormStyles.sectionTitle}>
+              {primary
+                ? t('profileInformation.primaryAddress')
+                : t('profileInformation.address')}
+            </h4>
+            <div className={commonFormStyles.multiItemWrapper}>
+              <Field
+                name="address"
+                id={`${testId}-address`}
+                maxLength={formFields.address.max as number}
+                as={TextInput}
+                invalid={hasFieldError(formikProps, 'address')}
+                aria-invalid={hasFieldError(formikProps, 'address')}
+                helperText={getFieldErrorMessage(formikProps, 'address')}
+                labelText={t(formFields.address.translationKey)}
+                autoFocus
+                aria-labelledby={`${dataType}-address-helper`}
+              />
+              <Field
+                name="postalCode"
+                id={`${testId}-postalCode`}
+                maxLength={formFields.postalCode.max as number}
+                as={TextInput}
+                invalid={hasFieldError(formikProps, 'postalCode')}
+                aria-invalid={hasFieldError(formikProps, 'postalCode')}
+                helperText={getFieldErrorMessage(formikProps, 'postalCode')}
+                labelText={t(formFields.postalCode.translationKey)}
+                aria-labelledby={`${dataType}-postalCode-helper`}
+              />
+              <Field
+                name="city"
+                id={`${testId}-city`}
+                maxLength={formFields.city.max as number}
+                as={TextInput}
+                invalid={hasFieldError(formikProps, 'city')}
+                aria-invalid={hasFieldError(formikProps, 'city')}
+                helperText={getFieldErrorMessage(formikProps, 'city')}
+                labelText={t(formFields.city.translationKey)}
+                aria-labelledby={`${dataType}-city-helper`}
+              />
+              <div className={commonFormStyles.formField}>
+                <FormikDropdown
```

Comment:
Could we apply the HDS DropDown component here? https://hds.hel.fi/components/dropdown

Combobox might fit better, since there's more than 8 options, and it might be quicker for the user to input first letter of the country and get the result?

Reply:
Currently you can type a letter and it jumps there. I don't know what HDS combobox is and when it was introduced. The designs do not use it.